pkg/codegen/internal/test/Makefile

```diff
@@ -0,0 +1,33 @@
+.PHONY = update_random update_aws update_azure update_azure_native update_kubernetes
+
+RANDOM_SCHEMA       := "https://raw.githubusercontent.com/pulumi/pulumi-random/master/provider/cmd/pulumi-resource-random/schema.json"
+AWS_SCHEMA          := "https://raw.githubusercontent.com/pulumi/pulumi-aws/master/provider/cmd/pulumi-resource-aws/schema.json"
+AZURE_SCHEMA        := "https://raw.githubusercontent.com/pulumi/pulumi-azure/master/provider/cmd/pulumi-resource-azure/schema.json"
+AZURE_NATIVE_SCHEMA := "https://raw.githubusercontent.com/pulumi/pulumi-azure-native/master/provider/cmd/pulumi-resource-azure-native/schema.json"
+KUBERNETES_SCHEMA   := "https://raw.githubusercontent.com/pulumi/pulumi-kubernetes/master/provider/cmd/pulumi-resource-kubernetes/schema.json"
+
+define add_version
+	cd testdata && gsed -i '3 i \ \ \ \ "version": "v$(2)",' $(1).json
+endef
+
+update_schema: update_random update_aws update_azure update_azure_native update_kubernetes
+
+update_random:
+	cd testdata && curl ${RANDOM_SCHEMA} > random.json
+	$(call add_version,random,4.2.0)
+
+update_aws:
+	cd testdata && curl ${AWS_SCHEMA} > aws.json
+	$(call add_version,aws,4.21.1)
+
+update_azure:
+	cd testdata && curl ${AZURE_SCHEMA} > azure.json
+	$(call add_version,azure,4.18.0)
+
+update_azure_native:
+	cd testdata && curl ${AZURE_NATIVE_SCHEMA} > azure-native.json
+	$(call add_version,azure-native,1.29.0)
+
+update_kubernetes:
+	cd testdata && curl ${KUBERNETES_SCHEMA} > kubernetes.json
```

Comment:
I'm still wondering if we depend on all the entire schemata or on some subset. I had a script lying around for extracting a subset of a schema by closing over references found in a type or two. It may be a legitimate strategy to depend on the entire schema but I'm thinking ahead to dependency loops between this package and the providers, there may be some slight downsides to that, some interesting maintenance situations where we can't build+ test one without a past version of the other.

Reply:
I think the solution to this is to depend only on pinned versions of the schema (which we need to do anyway). Upgrading the schema these tests depend on will need to be a manual process. As long as our providers exist and git keeps their release history, we should be fine. If we ever reach a point were we can't use the old schema, all these tests will need to be rewritten anyway, so I don't think we need to worry about it.

Reply:
>If we ever reach a point were we can't use the old schema, all these tests will need to be rewritten anyway

FWIW, this isn't true. We've updated the schema when we have added new features, and we certainly don't have to rewrite all the tests to update the schema version.

Reply:
Sorry @komalali, what I wrote was unclear. I meant if we ever made a breaking change to schemas, such that the current version of codegen is not compatible with the schemas currently used in the tests, we would need to rewrite the tests anyway.